### PR TITLE
B011_remove_Lombok_from_project

### PR DESCRIPTION
**Pull request by ilowski**
remove lombook dependency and data annotation from user model

### CODE REVIEW

File: src/main/java/it/hospital/project/appusers/User.java [hunk 1]
```diff
@@ -12,6 +12,7 @@
     @Id
```
Comment (rwedzony) [inline src/main/java/it/hospital/project/appusers/User.java:12]: We need Generation Strategy for ID generation

Comment (rwedzony) [inline src/main/java/it/hospital/project/appusers/User.java:12]: Hibernate requires no-args constructor to work.

File: src/main/java/it/hospital/project/service/UserService.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package it.hospital.project.service;
+
+import it.hospital.project.appusers.User;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
```
Comment (rwedzony) [inline src/main/java/it/hospital/project/service/UserService.java:7]: Do we plan to use somethere else UserService? Does it make sense to add additional level of abstraction?

File: src/main/java/it/hospital/project/controller/UserController.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+import it.hospital.project.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
```
Comment (rwedzony) [inline src/main/java/it/hospital/project/controller/UserController.java:11]: If you import already four annotations from org.springframework.web.bind.annotation you can already use "*" import

File: src/main/java/it/hospital/project/controller/UserController.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+@RestController
+@CrossOrigin
+@RequestMapping("/api/users")
+public class UserController {
+
+    UserService userService;
+
```
Comment (rwedzony) [inline src/main/java/it/hospital/project/controller/UserController.java:18]: There is no need to use autowired above constructor

File: src/main/java/it/hospital/project/controller/UserController.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@CrossOrigin
+@RequestMapping("/api/users")
```
Comment (rwedzony) [inline src/main/java/it/hospital/project/controller/UserController.java:14]: We structure application in domained way. So it means, you don't divide it by catalogues "services", "repositories", "controllers", but rather as domains, so "appusers" (where you put everything that is related to users, f.e models,services, controllers and so on), "medicalrecords". The structure you proposed becomes unefficient as project grows.

**Review by rwedzony — CHANGES_REQUESTED** [review]

File: src/main/java/it/hospital/project/service/UserService.java (hunk 2 above)
Comment (ilowski) [reply src/main/java/it/hospital/project/service/UserService.java:7]: You are asking about this interface? Its easier to check all methods which userServiceImpl is implementing

File: src/main/java/it/hospital/project/controller/UserController.java (hunk 5 above)
Comment (ilowski) [reply src/main/java/it/hospital/project/controller/UserController.java:14]: Can i create catalog "appusers" and then put there other catalogues such as "services" "repositories" or should I upload only file, classes?

**Pull request closed without merging**